src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,116 @@
 package co.inventorsoft.academy.collections.model;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.List;
 import java.util.Set;
+import java.util.function.Function;
+
 
 public class Range<T> implements Set<T> {
 
+    List<T> mylist = new ArrayList();
+
+    Range() {
+    }
+
+    public static Range of(int i, int i1) {
+
+        Range<Integer> intRange = Range.of(i, i1, i2 -> (int) (i2 + 1));
+        return intRange;
+    }
+
+    public static Range of(float i, float i1) {
+        Range<Float> floatRange = Range.of(i, i1, i2 -> (float) (i2 + 0.1));
+        return floatRange;
+
+    }
+
+
     public int size() {
-        return 0;
+        return mylist.size();
     }
 
     public boolean isEmpty() {
-        return false;
+        return mylist.isEmpty();
     }
 
+    @Override
     public boolean contains(Object o) {
-        return false;
+        return mylist.contains(o);
     }
 
-    public Iterator<T> iterator() {
-        return null;
+    @Override
+    public Iterator iterator() {
+        return mylist.iterator();
     }
 
+    @Override
     public Object[] toArray() {
-        return new Object[0];
+        return mylist.toArray();
     }
 
-    public <T1> T1[] toArray(T1[] a) {
-        return null;
+    @Override
+    public boolean remove(Object o) {
+        return mylist.remove(o);
     }
 
-    public boolean add(T t) {
+    public boolean addAll(Collection<? extends T> c) {
+        for (T t : mylist) {
+            return add(t);
```

Comment:
I don't think that method addAll works as expected. It adds one element and returns.

Reply:
fix